vagrant.go

```diff
@@ -174,6 +187,20 @@ func (v *Vagrant) Setup(start bool, env string, numNodes int) error {
 	return nil
 }
 
+// Setup initializes a vagrant testbed.
+func (v *Vagrant) Setup(args ...interface{}) error {
```

Comment:
hmmm, just to make sure I understood correctly, basically do something as in a code sketch  below:

```
type SetupConfig struct {
vagrant VagrantConfig
baremetal BaremetalConfig
... 
}

func (v *vagrant) Setup (c *SetupConfig) error {
    return v.setup(c.vagrant)
}

...
...

func (bm *baremetal) Setup (c *SetupConfig) error {
    return bm.setup(c.baremetal)
}

```
I thought about it, it works too and the cast related errors in the `Setup()` function actually goes away because we don't have much validations there today. The thing I like about current organization is that the setup functions of different testbed type are pretty independent and have their own validations.

A common config structure like above kind of mixes the setup fields for different testbeds and that the client code still needs to understand (than just understanding the options for their testbed). But it is minor. 

Also a little minor, the common config structure approach will break the current clients (like volplugin).

I am fine moving to a combined config structure if that still looks better.